js/serial.js

```diff
@@ -18,18 +18,11 @@ var serial = {
     logHead: 'SERIAL: ',
 
     connect: function (path, options, callback) {
-        var self = this;
```

Comment:
Why did you remove this? I think as a pattern in JS it's useful to take a reference to this at the beginning of the function, and then use the reference to refer to this. This avoids errors when refactoring is done and stuff is moved into callbacks later on.